[sBTC] Add stx_address method to PegOutRequestOp

netrome opened the pull request:
This PR adds a `PegOutRequestOp::stx_address()` method to derive the stacks address of an sBTC holder requesting to peg out.

Closes https://github.com/stacks-network/stacks-blockchain/issues/3647

igorsyl commented:
@kantai I would be grateful for your attention to this PR given it's one of the last remaining pieces for a working sBTC Alpha.

![image](https://user-images.githubusercontent.com/1248500/230699752-b9551eec-6422-4a24-85c3-46433637712a.png)

kantai reviewed (commented):
This LGTM, but it needs two more kinds of tests:

1. A test that uses a test vector (i.e., just a hex encoded bitcoin transaction) as input, with a hard coded expected `stx_address`.
2. Test coverage for the two error cases

netrome commented:
> This LGTM, but it needs two more kinds of tests:
> 
>     1. A test that uses a test vector (i.e., just a hex encoded bitcoin transaction) as input, with a hard coded expected `stx_address`.
> 
>     2. Test coverage for the two error cases

Thank you for the feedback. I have added a test case with the test vector and another one for one of the error cases. I discovered that the other error case was infallible, so I changed the method and return type to reflect that.

It would be great if you could have another look @kantai :pray:

kantai reviewed (approved):
LGTM!

igorsyl commented:
Thank you @kantai

blockstack-devops commented:
This pull request has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.
